game_board.py

```diff
@@ -0,0 +1,199 @@
+
+        board_txt = board.read()
+        return board_txt
+    except IOError:
+        print("Can not find the file" )
+
+
+
+
+def map_cord(str_board):
+    """
+    Reads the string from left to right and appends every char to list with its x any y pos
+    the top of the menu. 
+    Returns a list of lists where every list has [char,xPos,Ypos]
+    Example "X  " => [['X', 0, 0], [' ', 1, 0], [' ', 2, 0]...,]
+
+    Keyword arguments:
+    str_board -- the string of the map
+    """
+ 
+    matrix = [[x for x in line] for line in str_board.split('\n')]
+    map_xy = []
+    y = 0
+    for row in matrix:
+        x = 0
+        for c in row:
+            char_xy = [c,x,y]    
+            map_xy.append(char_xy)
+            x += 1
+        y += 1
+    return map_xy
+
+def move_plus():
+    """
+    Finds all the "+" chars in the map_xy 
+    Returns a list of lists where every list has ["+",xPos,Ypos]
+    Example [['X', 0, 0], [' ', 1, 0],...,['+', '4', '0']] => [['+', '4', '0'],...]
+    """
+    map_xy =map_cord(read_map(map_path))
+    move = []
+    for cord in map_xy:
+        if cord[0] == "+":
+            move.append([cord[0],str(cord[1]),str(cord[2])])
+    return move
+
+def print_map(screen):
+    """Reads the map_xy list  and prints the chars in the Screen
+
+    Keyword arguments:
+    screen -- the curses screen
+    map_xy -- list of lists the chars and their positions 
+    """
+    map_xy = map_cord(read_map(map_path))
+    for cord in map_xy:
+        h,w = screen.getmaxyx()
+        y = cord[2]+ 5
+        x = cord[1] +  w//3    
+        char = cord[0]
+        if char == "O":
+            curses.init_pair(2,curses.COLOR_YELLOW,curses.COLOR_BLACK)
+                        
+            screen.addstr(y,x,char,curses.color_pair(2))
+        elif char == "X": 
+            curses.init_pair(3,curses.COLOR_RED,curses.COLOR_BLACK)
+
+            screen.addstr(y,x,char,curses.color_pair(3))
+        elif char != "+" and char != "O" and char !="X":
+            screen.addstr(y,x,char)
+
+      
+          
+
+
+def print_choice(screen,selected_move_idx,move):
+    """Reads the move list  and prints the chars in the Screen. If selected
+
+    Keyword arguments:
+    screen -- the curses screen
+    selected_move_idx -- the currently selected row in the move
+    move -- the lists of all the "+" and their positions
+    """    
+    screen.clear()
+    print_map(screen)
+    h, w = screen.getmaxyx()
+    for idx, row in enumerate(move):
+        y = int(row[2])+ 5
+        x = int(row[1]) +  w//3      
+        if idx == selected_move_idx:
+            screen.attron(curses.color_pair(1))
+            screen.addstr(y,x, row[0])
+            screen.attroff(curses.color_pair(1))
+        else:
+            screen.addstr(y, x, row[0])
+   
+    screen.refresh()    
+
+
+def move_down(move,current_row):
+    """ Finds the "+" char which has the same xpos but an other Ypos in the move list
+    return the index of the new "+"
```

Comment:
```suggestion
    Returns the y position of the new "+"
```

Reply:
The func returns New_row which is an index in plus_list

Reply:
Yes and would you agree that the index represents the y position of the new "+"? Because I find the index to be a bit unspecific of what it represents. When I read the description I get uncertain if it represent a position in both the x and y axis of our map or just the x axis or just the y axis.